public/models/Player.js

```diff
@@ -12,39 +12,123 @@ class Player extends GameObject { // Marianna
+          // console.log('check row', row - 1, 'column', column + 1);
+        }
+        break;
+      case 'down':
+        if (y % map.tileHeight) {
+          collision = this.handleCollision(map.tiles[row][column], row, column, onlyWallCollision);
+          // console.log('check row', row , 'column', column);
+          if (x % map.tileWidth) {
+            collision = this.handleCollision(map.tiles[row][column + 1], row, column + 1, onlyWallCollision) || collision;
+            // console.log('check row', row, 'column', column + 1);
+          }
+        } else {
+          collision = this.handleCollision(map.tiles[row - 1][column], row - 1, column, onlyWallCollision);
+          // console.log('check row', row - 1, 'column', column);
+          if (x % map.tileWidth) {
+            collision = this.handleCollision(map.tiles[row - 1][column + 1], row - 1, column + 1, onlyWallCollision) || collision;
+            // console.log('check row', row - 1, 'column', column + 1);
+          }
+        }
+        break;
+      case 'right':
+        if (x % map.tileWidth) {
+          collision = this.handleCollision(map.tiles[row - 1][column + 1], row - 1, column + 1, onlyWallCollision);
+          // console.log('check row', row - 1, 'column', column + 1);
+          if (y % map.tileHeight) {
+            collision = this.handleCollision(map.tiles[row][column + 1], row, column + 1, onlyWallCollision) || collision;
+            // console.log('check row', row, 'column', column + 1);
+          }
+        } else {
+          collision = this.handleCollision(map.tiles[row - 1][column], row - 1, column, onlyWallCollision);
+          // console.log('check row', row - 1, 'column', column);
+          if (y % map.tileHeight) {
+            collision = this.handleCollision(map.tiles[row][column], row, column, onlyWallCollision) || collision;
+            // console.log('check row', row, 'column', column + 1);
+          }
+        }
+        break;
+      case 'left':
+        collision = this.handleCollision(map.tiles[row - 1][column], row - 1, column, onlyWallCollision);
+        // console.log('check row', row - 1, 'column', column);
+        if (y % map.tileHeight) {
+          collision = this.handleCollision(map.tiles[row][column], row, column, onlyWallCollision) || collision;
+          // console.log('check row', row, 'column', column);
+        }
+        break;
+      default:
     }
+    return collision;
+  }
+
+  handleCollision(block, row, column, onlyWallCollision) {
+    // check only wall collisions
+    if (block !== 1 && onlyWallCollision) block = '';
+    block = String(block);
+    switch (block) {
+      case '1':
+        // console.log("wall", row, column);
+        return true;
+      case '2':
+        // console.log("goal", row, column);
+        this.playerIsDone();
+        break;
+      case (block.match(/^4/)?.input):
```

Comment:
;block is col and row? what does it mean : block.match(/^4/)?.input

Reply:
Block is basically type of the block in tilesheet. As there is used 0 for path, 1 for wall and so on. But for coins it is a bit tricky because you have coins with different values, so you can't just have there 4 on the map. Coins are all in an array so to differentiate between them it is written like 4.0 4.1 where the 2nd number is position in an array. In this case it is just checking whether the number at the specific column/row starts with 4 to see whether it is coin.